src/mds/MDLog.cc

```diff
@@ -626,7 +626,14 @@ void MDLog::trim(int m)
     op_prio = CEPH_MSG_PRIO_HIGH;
 
   unsigned new_expiring_segments = 0;
-
+  
+  unsigned expiring_segments_size = g_conf().get_val<uint64_t>("mds_log_max_expiring_segments");
+  if(segments.size() > pre_segments_size && expiring_segments_size > 0)
+    expiring_segments_size = std::max(expiring_segments_size,static_cast<unsigned>(segments.size() - pre_segments_size));
```

Comment:
this code limits expiring segments when segments.size() <= pre_segments_size. This is not expected